Dev 4.2

maurar3 opened the pull request:
Made minor changes to template code for 4.2 release, including:

 Changed filename listViewModel.js t dataViewModel.js
 Updated id names in various html files
 Made minor modifications due to running JSHint
 Updated text in ReadMe.txt file

maurar3 committed:
4.2 Release

maurar3 committed:
4.2 Release

maurar3 committed:
4.2 Release

maurar3 committed:
4.2 Release

schase100 committed:
Update README.md

maurar3 committed:
Update loginViewModel.js

Donna McMann found issue where LogIn View's title was not being updated

maurar3 committed:
Update README.md

Updating to correct repository uri for branch

maurar3 committed:
Update README.md

Updating branch repository URI round 2

maurar3 committed:
Update README.md

maurar3 committed:
Update README.md

maurar3 committed:
Found issue with title using anonymous authModel

KristianDD commented on README.md:
```diff
@@ -4,7 +4,7 @@ Mobile Project Template
 A project template for creating mobile applications designed for use with a Progress Data Service (OpenEdge or Rollbase).
 
-<a href="https://platform.telerik.com/#appbuilder/clone/https%3A%2F%2Fgithub.com%2FCloudDataObject%2Fappbuilder-mobile-template" target="_blank"><img src="http://docs.telerik.com/platform/appbuilder/sample-apps/images/try-in-appbuilder.png" alt="Try in AppBuilder" title="Try in AppBuilder" /></a>
+<a href="https://platform.telerik.com/#appbuilder/clone/https%3A%2F%2Fgithub.com%2FCloudDataObject%2Fappbuilder-mobile-template%2Ftree%2Fdev_4.2" target="_blank"><img src="http://docs.telerik.com/platform/appbuilder/sample-apps/images/try-in-appbuilder.png" alt="Try in AppBuilder" title="Try in AppBuilder" /></a>
```
Maybe this should be returned to its original "href" before merge in master, because the way it is now, when the Pull Request gets merged,  users will be cloning from "dev_4.2" branch.

maurar3 replied:
Yes, agreed

KristianDD commented on mobile template/viewModels/dataViewModel.js:
```diff
@@ -1,10 +1,10 @@
-'use strict';
+//'use strict';
```
Maybe the line should be deleted or should remove comment.

maurar3 replied:
This was intentionally left in, to show customer option of using it

KristianDD commented on mobile template/viewModels/loginViewModel.js:
```diff
@@ -1,24 +1,27 @@
-'use strict';
+//"use strict";
```
Maybe the line should be deleted or should remove comment.

maurar3 replied:
This was intentionally left in, to show customer option of using it

KristianDD commented on mobile template/viewModels/loginViewModel.js:
```diff
@@ -1,24 +1,27 @@
+        loginViewTitle: "Log In",
+        loginLabel: "Log In",
+        logoutLabel: "Log Out",
+  
```
There are white spaces on the empty line.

maurar3 replied:
Will remove

KristianDD commented on mobile template/viewModels/loginViewModel.js:
```diff
@@ -1,24 +1,27 @@
+            if (!app.isAnonymous()) {
+                app.changeTitle(app.viewModels.loginViewModel.loginViewTitle);
+            }
+          
```
There are white spaces on the empty line.

maurar3 replied:
Will remove

KristianDD commented on mobile template/viewModels/loginViewModel.js:
```diff
@@ -43,7 +45,9 @@
                         that.set("isLoggedIn", true);
+                        app.viewModels.loginViewModel.loginViewTitle = app.viewModels.loginViewModel.logoutLabel;
                         app.viewModels.loginViewModel.onBeforeShow( );
+                        
```
There are white spaces on the empty line.

maurar3 replied:
Will remove

KristianDD commented on mobile template/viewModels/loginViewModel.js:
```diff
@@ -68,15 +72,15 @@
             }
             catch(ex) {
-               app.viewModels.loginViewModel.loginErrorFn(app.jsdosession, progress.data.Session.GENERAL_FAILURE, 
+               app.viewModels.loginViewModel.loginErrorFn(app.jsdosession, 
```
Trailing white space at the end of the line and on next line.

maurar3 replied:
Will remove

KristianDD commented on mobile template/viewModels/loginViewModel.js:
```diff
@@ -86,11 +90,12 @@
+                    if (app.viewModels.dataViewModel) {
                         // Remove any leftover data
-                        app.viewModels.listViewModel.clearData();   
+                        app.viewModels.dataViewModel.clearData();   
```
Trailing white space at the end of the line.

maurar3 replied:
Will remove

KristianDD commented on mobile template/viewModels/loginViewModel.js:
```diff
@@ -98,8 +103,9 @@
             catch(ex) {
-               app.viewModels.loginViewModel.logoutErrorFn(app.jsdosession, progress.data.Session.GENERAL_FAILURE, 
-                   {errorObject: ex});
+               app.viewModels.loginViewModel.logoutErrorFn(app.jsdosession, 
```
Trailing white space at the end of the line and on next line.

maurar3 replied:
Will remove

KristianDD commented on mobile template/viewModels/loginViewModel.js:
```diff
@@ -117,7 +123,8 @@
             if (details !== undefined  && Array.isArray(details)){
                 for (i = 0; i < details.length; i += 1){
-                    msg = msg + "\nresult for " + details[i].catalogURI + ": " + details[i].result + "\n    " + details[i].errorObject;
+                    msg = msg + "\nresult for " + details[i].catalogURI + ": " + 
```
Trailing white space at the end of the line.

maurar3 replied:
Will remove

KristianDD commented on mobile template/viewModels/loginViewModel.js:
```diff
@@ -143,15 +150,14 @@
-                    msg = msg + " Service " + jsdoSettings.serviceURI + " is unavailable";
-                    break;
-            }       
+            
```
There are white spaces on the empty line.

maurar3 replied:
Will remove

KristianDD commented on mobile template/viewModels/loginViewModel.js:
```diff
@@ -143,15 +150,14 @@
+            else {
+                msg = msg + " Service " + jsdoSettings.serviceURI + " is unavailable";
+            }
+                  
```
There are white spaces on the empty line.

maurar3 replied:
Will remove

maurar3 committed:
Update README.md

Reverting back to text in main branch. These changes were only done for testing purposes, and need to be backed out

maurar3 committed:
Removing extra spaces, based upon comments from KristianDD

maurar3 committed:
Removing extra spaces, based upon comments from KristianDD

maurar3 committed:
Update README.md

maurar3 committed:
Update progress.all.js

Updated to latest 4.2 version

maurar3 committed:
Updated to latest 4.2 Version

maurar3 committed:
Update README.md

maurar3 committed:
Updated Readme.md

License link was not working

maurar3 committed:
4.2 Release

maurar3 commented:
I wound up having to do a manual merge of dev_4.2 up to the main branch due to conflict in README.md
